### DISCUSSION TITLE
[@types/react] Why React.FunctionComponent returns ReactElement instead of ReactNode?

### DISCUSSION THREAD

**Discussion by Rogach** [DefinitelyTyped/DefinitelyTyped#53846] [Issues with a @types package]
Example component to showcase the issue:

```typescript
export const TestFnComponent: React.FunctionComponent = props => {
  return "deadbeef";
};
```

TypeScript throws an error:
```
TS2322: Type '(props: { children?: ReactNode; }) => string' is not assignable to type 'FunctionComponent<{}>'.
Type 'string' is not assignable to type 'ReactElement<any, any>'.
```

The issue is because FunctionComponent is defined as returning `ReactElement<any, any> | null`, and ReactElement does not include `string` in possible values.

`ReactNode` seems to be more appropriate here - why React.FunctionComponent opts to return ReactElement instead?

**Comment by eugene-matvejev** [comment]
it seems like if you do ` as React.ReactNode & React.ReactElement`

e.g. 
`const FN : React.FC = () => { return "abc"  as React.ReactNode & React.ReactElement }`
should work. `<>{"abc"}</>` it is adding extra code in production build. while my is typing only, but it is weird and nasty I must agree

**Comment by msakrejda** [comment]
Here's an older issue discussing this problem: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20544 

Changing the type declaration is a workaround, but you may already have code (or be working with library code) that does something like this:

```tsx
function MyFancyLayout({
  content: Content,
  sidebar: Sidebar,
}: {
  content React.ComponentType;
  sidebar: React.ComponentType;
}) {
  return (
    <div><Content/><Sidebar/></div>
  )
}
```

TypeScript will complain if this gets passed a `string` as content, which seems silly.

I think the solution is being blocked by https://github.com/microsoft/TypeScript/issues/21699 but that's a huge thread and I have a hard time telling what the status there is.
